src/strategies/reverse-voting-escrow/test.ts

```diff
@@ -0,0 +1,205 @@
+        ]);
+      }
+    }
+
+    // ** Execute the vested multicall ** //
+    try {
+      // This should return a mapping of wallet addresses to vested amounts
+      const tempWalletVestedAmounts: Record<
+        string,
+        typeof BigNumber
+      > = await getWalletToVestedAmount.execute();
+      console.log('Got temp wallet vested amount:', tempWalletVestedAmounts);
+      walletToVestedAmount = Object.assign(
+        walletToVestedAmount,
+        tempWalletVestedAmounts
+      );
+    } catch (e) {
+      console.log('multicall errored!');
+      console.log(e);
+      // !! IGNORE Multicall REVERTS !! //
+    }
+  }
+
+  // ** Execute the claimed multicall ** //
+  const walletToClaimedAmount: Record<
+    string,
+    typeof BigNumber
+  > = await getWalletToClaimedAmount.execute();
+
+  // ** Map address to its full amount of claimable tokens ** //
+  const listOfFullAmounts = Object.entries(allDataJSON)
+    .map(([cohortId, cohortData]: any) =>
+      Object.entries(cohortData).map(([address, data]: any) => [
+        cohortId,
+        address,
+        data.amount,
+        data.index
+      ])
+    )
+    .reduce((prev, curr) => {
+      return prev.concat(curr);
+    });
+  const fullAmounts: Map<string, typeof BigNumber> = new Map(
+    listOfFullAmounts.map(([, address, amount]) => [address, amount])
+  );
+  console.log('Iterating over full amounts...');
+
+  // ** Iterate over the full amounts mapping from address -> total claim amount **//
+  fullAmounts.forEach((amount, address) => {
+    const vestedBalance: typeof BigNumber =
+      walletToVestedAmount[address] || BigNumber.from(0);
+    console.log('Address', address, 'has vestedBalance:', vestedBalance);
+    const claimedBalance: typeof BigNumber =
+      walletToClaimedAmount[address] || BigNumber.from(0);
+    console.log('Address', address, 'has claimedBalance:', claimedBalance);
+
+    // ?? voting_power = current_balance +
+    // ??                vested_balance +
+    // ??                0.1 * (fullAmount - vestedBalance - claimedBalance)
+
+    console.log('Address', address, 'has amount:', BigNumber.from(amount));
+    const vestingPower = BigNumber.from(amount)
+      .sub(vestedBalance)
+      .sub(claimedBalance)
+      .div(BigNumber.from(10));
+    console.log('Vesting power:', vestingPower);
+    const addedVotingPower = vestedBalance.add(vestingPower);
+    console.log('Added Voting Power:', addedVotingPower);
+    reverseVotingBalance[address] = reverseVotingBalance[address]
+      ? reverseVotingBalance[address].add(addedVotingPower)
+      : addedVotingPower;
+  });
+
+  // ** Return [address, balance] pairs ** //
+  const finalReturnData = Object.fromEntries(
+    addresses.map((address) => [
+      address,
+      reverseVotingBalance[address]
+        ? parseFloat(reverseVotingBalance[address].toString())
+        : 0
+    ])
+  );
+  console.log('Final return data', finalReturnData);
+};
+
+main();
+
+export {};
```

Comment:
Hi @abigger87 why we need this file? you can run this command to test the strategy
```
yarn test --strategy=reverse-voting-escrow 
```